### PR TITLE
feat(comment): Create comment action widgets

### PR DESCRIPTION
**Pull request by andre-tm-hui**
Motivation: add widgets for comment like, flag and delete, and the appropriate popup dialogs.
Relates: #175 

Widget demos:
[commentwidgets.webm](https://user-images.githubusercontent.com/34312863/204622576-81102b5a-321c-4bed-9898-4899059792eb.webm)
The specific dialog for when a report is sent:
[flaggeddialog.webm](https://user-images.githubusercontent.com/34312863/204622733-cde5149d-2b13-4f64-a431-e85e86be1830.webm)

### CODE REVIEW

File: lib/presentation/crowdaction/crowdaction_comments/widgets/comment_delete_button.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import 'package:flutter/material.dart';
+
+///import '../../../core/collaction_icons.dart';
```
Comment (Xazin) [inline lib/presentation/crowdaction/crowdaction_comments/widgets/comment_delete_button.dart:3]: Remove

File: lib/presentation/crowdaction/crowdaction_comments/widgets/comment_delete_button.dart [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  const CommentDeleteButton({
+    super.key,
+    this.onTap,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return IconButton(
+      /// TODO: replace icons with variants from CollactionIcons
```
Comment (Xazin) [inline lib/presentation/crowdaction/crowdaction_comments/widgets/comment_delete_button.dart:16]: We use IonIcons, you should be able to find it, it's in the project most likely just not used anywhere.

File: lib/presentation/crowdaction/crowdaction_comments/widgets/comment_flag_button.dart [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import 'package:flutter/material.dart';
+
+///import '../../../core/collaction_icons.dart';
```
Comment (Xazin) [inline lib/presentation/crowdaction/crowdaction_comments/widgets/comment_flag_button.dart:3]: Remove

File: lib/presentation/crowdaction/crowdaction_comments/widgets/comment_like_button.dart [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import 'package:flutter/material.dart';
+
+///import '../../../core/collaction_icons.dart';
```
Comment (Xazin) [inline lib/presentation/crowdaction/crowdaction_comments/widgets/comment_like_button.dart:3]: Remove

File: lib/presentation/crowdaction/crowdaction_comments/widgets/flag_comment.dart [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+    super.key,
+    this.flagged = false,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    /// to be replaced with a BlocBuilder to swap between dialog and success
+    if (!flagged) {
+      return FlagDialog(
+        /// need clarification - will these be defined/accessible in the API? do we need a flag obj?
```
Comment (Xazin) [inline lib/presentation/crowdaction/crowdaction_comments/widgets/flag_comment.dart:21]: Good point, let's refactor this later, it's fine as it is right now.

**Review by Xazin — DISMISSED** [review]
Your work is very straightforward and relies on default behavior, love it.

File: lib/presentation/crowdaction/crowdaction_comments/widgets/comment_delete_button.dart (hunk 1 above)
Comment (andre-tm-hui) [reply lib/presentation/crowdaction/crowdaction_comments/widgets/comment_delete_button.dart:3]: Done

File: lib/presentation/crowdaction/crowdaction_comments/widgets/comment_delete_button.dart (hunk 2 above)
Comment (andre-tm-hui) [reply lib/presentation/crowdaction/crowdaction_comments/widgets/comment_delete_button.dart:16]: I've changed it now, though I couldn't find a similar flagging icon to the one in the figma, so I've used Ionicons.flag. Is that alright for now?

Comment (Xazin) [reply lib/presentation/crowdaction/crowdaction_comments/widgets/comment_delete_button.dart]: Yes that is what I meant 👍

**Review by Xazin — APPROVED** [review]

**Comment by andre-tm-hui** [comment]
Thanks! There were lots of examples in the code base, so I just used them as reference.

**Pull request merged**